tests/evolution/test_evaluation_context.py

```
@@ -0,0 +1,246 @@
+This module tests the new evaluation context functionality that passes
+original question and scoring information to semantic operators.
+"""
+
+import pytest
+from unittest.mock import AsyncMock, MagicMock
+from dataclasses import asdict
+
+from mad_spark_alt.evolution.interfaces import (
+    EvolutionRequest, EvolutionConfig, IndividualFitness, EvaluationContext
+)
+from mad_spark_alt.evolution.genetic_algorithm import GeneticAlgorithm
```

Ruff (0.12.2)

`mad_spark_alt.evolution.genetic_algorithm.GeneticAlgorithm` imported but unused

Remove unused import: `mad_spark_alt.evolution.genetic_algorithm.GeneticAlgorithm`

(F401)